docs/sources/mimir/configure/configuration-parameters/index.md

```diff
@@ -4622,6 +4622,10 @@ The s3_backend block configures the connection to Amazon S3 object storage backe
 # CLI flag: -<prefix>.s3.native-aws-auth-enabled
 [native_aws_auth_enabled: <boolean> | default = false]
 
+# (experimental) Whether or not to disable multipart updloads.
+# CLI flag: -<prefix>.s3.disable-multipart
+[disable_multipart: <boolean> | default = false]
```

Comment:
Can you explain your use case for disabling multipart uploads? Is it a compatibility concern? If so, which object storage service is it for?

Reply:
we discovered that we could achieve better GET performances from our internal objecstorage with multipart disabled.

our use case is a high load with a high frequency (1s interval). We were trying to play with the part size while we discovered it was not possible do disable it all together (thanos-io/objstore [resets part size if the object size is higher than the part size](https://github.com/thanos-io/objstore/blob/main/providers/s3/s3.go#L485C16-L486C1)).

Moreover all S3 clients have an option to disable multi-part uploads. Not having the option in mimir looks like something was missing.

Reply:
Interesting, thanks for the explanation. It sounds like you're benefiting from some kind of data colocation. Downside of disabling multipart is higher write latency (which Mimir isn't sensitive to) and being more vulnerable to write retries. That being said, exposing the option seems reasonable enough to me if it's added to upstream.

Reply:
it's been added to upstream and bot already updated module version with the lastest version.